v2/webhooks.go

```diff
@@ -0,0 +1,152 @@
+	WebhookDiscordProviderType    WebhookProviderType = "discord"
+)
+
+const (
+	WebhookNodeCreated                    WebhookSubscriptionType = "nodeCreated"
+	WebhookNodeNeedsApproval              WebhookSubscriptionType = "nodeNeedsApproval"
+	WebhookNodeApproved                   WebhookSubscriptionType = "nodeApproved"
+	WebhookNodeKeyExpiringInOneDay        WebhookSubscriptionType = "nodeKeyExpiringInOneDay"
+	WebhookNodeKeyExpired                 WebhookSubscriptionType = "nodeKeyExpired"
+	WebhookNodeDeleted                    WebhookSubscriptionType = "nodeDeleted"
+	WebhookPolicyUpdate                   WebhookSubscriptionType = "policyUpdate"
+	WebhookUserCreated                    WebhookSubscriptionType = "userCreated"
+	WebhookUserNeedsApproval              WebhookSubscriptionType = "userNeedsApproval"
+	WebhookUserSuspended                  WebhookSubscriptionType = "userSuspended"
+	WebhookUserRestored                   WebhookSubscriptionType = "userRestored"
+	WebhookUserDeleted                    WebhookSubscriptionType = "userDeleted"
+	WebhookUserApproved                   WebhookSubscriptionType = "userApproved"
+	WebhookUserRoleUpdated                WebhookSubscriptionType = "userRoleUpdated"
+	WebhookSubnetIPForwardingNotEnabled   WebhookSubscriptionType = "subnetIPForwardingNotEnabled"
+	WebhookExitNodeIPForwardingNotEnabled WebhookSubscriptionType = "exitNodeIPForwardingNotEnabled"
+)
+
+type (
+	// WebhookProviderType defines the provider type for a Webhook destination.
+	WebhookProviderType string
+
+	// WebhookSubscriptionType defines events in tailscale to subscribe a Webhook to.
+	WebhookSubscriptionType string
+
+	// Webhook type defines a webhook endpoint within a tailnet.
+	Webhook struct {
+		EndpointID       string                    `json:"endpointId"`
+		EndpointURL      string                    `json:"endpointUrl"`
+		ProviderType     WebhookProviderType       `json:"providerType"`
+		CreatorLoginName string                    `json:"creatorLoginName"`
+		Created          time.Time                 `json:"created"`
+		LastModified     time.Time                 `json:"lastModified"`
+		Subscriptions    []WebhookSubscriptionType `json:"subscriptions"`
+		// Secret is only populated on Webhook creation and after secret rotation.
+		Secret *string `json:"secret,omitempty"`
+	}
+
+	// CreateWebhookRequest type describes the configuration for creating a Webhook.
+	CreateWebhookRequest struct {
+		EndpointURL   string                    `json:"endpointUrl"`
+		ProviderType  WebhookProviderType       `json:"providerType"`
+		Subscriptions []WebhookSubscriptionType `json:"subscriptions"`
+	}
+)
+
+type WebhooksResource struct {
+	*Client
+}
+
+// Create creates a new webhook with the specifications provided in the CreateWebhookRequest.
+// Returns a Webhook if successful.
+func (c *WebhooksResource) Create(ctx context.Context, request CreateWebhookRequest) (*Webhook, error) {
```

Comment:
`Webhook` has been removed from function names since it's already named in the Resource.